services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -6902,11 +6899,21 @@ Store checkPreConditionForAclOp(String clusterName, String storeName) {
     return store;
   }
 
+  /**
+   * A store is not hybrid in the following two scenarios:
+   * If hybridStoreConfig is null, it means store is not hybrid.
+   * If all the hybrid config values are negative, it indicates that the store is being set back to batch-only store.
+   */
   boolean isHybrid(HybridStoreConfig hybridStoreConfig) {
-    /** A store is not hybrid in the following two scenarios:
-     * If hybridStoreConfig is null, it means store is not hybrid.
-     * If all of the hybrid config values are negative, it indicates that the store is being set back to batch-only store.
-     */
+    return hybridStoreConfig != null
+        && (hybridStoreConfig.getRewindTimeInSeconds() >= 0 || hybridStoreConfig.getOffsetLagThresholdToGoOnline() >= 0
+            || hybridStoreConfig.getProducerTimestampLagThresholdToGoOnlineInSeconds() >= 0);
+  }
+
```

Comment:
Duplicate codes.
Can we construct a `HybridStoreConfigImpl` out of `HybridStoreConfigRecord` and reuse the same comparison logic?

Reply:
Sure